osf/migrations/0161_add_registration_files_count.py

```diff
@@ -0,0 +1,50 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.15 on 2019-01-18 14:43
+from __future__ import unicode_literals
+import logging
+import progressbar
+
+from django.db import migrations, models
+from django.db.models import Count
+from django_bulk_update.helper import bulk_update
+from osf.models import Registration
```

Comment:
importing Registration outside, that way I have access to the files relationship to do everything I need. This was the only way I've found to see a significant speed increase so far.

Reply:
Comment added - 

When you import the model inside the function (which I'm not doing here), you're dealing with the historical model, the model as it was at that time, but it doesn't have all the managers, mixins, etc. associated with it.  So this migration is usually better at standing the test of time, but you lose a lot of the fancy things you normally can utilize on the model in your migration.
```python
Registration = state.get_model('osf', 'Registration')
```

If the Registration model changes in a way that will make this migration fail, travis will fail when it runs migrations and we can fix it at the time.  This happens occasionally, other migrations have made similar tradeoffs.